geode-core/src/main/java/org/apache/geode/internal/net/BufferPool.java

```diff
@@ -114,6 +138,70 @@ private ByteBuffer acquireBuffer(int size, boolean send) {
     return result;
   }
 
+  private ByteBuffer acquireSmallBuffer(boolean send) {
+    // set
+    int size = Connection.SMALL_BUFFER_SIZE;
```

Comment:
I don't think this size management in this method (or the next middle one) is correct. I think you need to pass the size in. When we find an existing one to use then we want to set its limit to the size that was requested, not to SMALL_BUFFER_SIZE. But if we do not find one then we want to allocate a new one of size SMALL_BUFFER_SIZE (and then set its limit to "size" which can be <= SMALL_BUFFER_SIZE).